src/meshTools/AMIInterpolation/AMIInterpolation/AMIMethod/directAMI/directAMI.C

```diff
@@ -232,96 +403,107 @@ void Foam::directAMI<SourcePatch, TargetPatch>::calculate
     label tgtFaceI
 )
 {
-    bool ok =
-        this->initialise
-        (
-            srcAddress,
-            srcWeights,
-            tgtAddress,
-            tgtWeights,
-            srcFaceI,
-            tgtFaceI
-        );
+    this->checkPatches();
+
+    // set initial sizes for weights and addressing
+    srcAddress.setSize(this->srcPatch_.size());
+    srcWeights.setSize(this->srcPatch_.size());
+    tgtAddress.setSize(this->tgtPatch_.size());
+    tgtWeights.setSize(this->tgtPatch_.size());
 
-    if (!ok)
+    // check that patch sizes are valid
+    if (!this->srcPatch_.size())
     {
         return;
     }
+    else if (!this->tgtPatch_.size())
+    {
+        WarningIn
+            (
+                "void Foam::AMIMethod<SourcePatch, TargetPatch>::initialise"
+                    "("
+                    "labelListList&, "
+                    "scalarListList&, "
+                    "labelListList&, "
+                    "scalarListList&, "
+                    "label&, "
+                    "label&"
+                    ")"
+            )
+        << this->srcPatch_.size() << " source faces but no target faces" << endl;
 
+        return;
+    }
 
-    // temporary storage for addressing and weights
-    List<DynamicList<label> > srcAddr(this->srcPatch_.size());
-    List<DynamicList<label> > tgtAddr(this->tgtPatch_.size());
+    const pointgpuField& tgtPoints = this->tgtPatch_.getPoints();
+    const pointgpuField& srcPoints = this->srcPatch_.getPoints();
```

Comment:
Arrays which hold the number of intersecting faces for each source and target face.